metrics/metrics_test.go

```diff
@@ -161,3 +163,97 @@ func TestMetricsCollectorByOrg(t *testing.T) {
 
 	require.Equal(t, uint64(999), total)
 }
+
+type closeDelayCase struct {
+	name       string
+	immediate  bool
+	syncTime   time.Duration
+	delay      time.Duration
+	expCounter int
+}
+
+func TestClosesWithDelay(t *testing.T) {
+	for _, c := range []closeDelayCase{
+		{"ImmediateWithoutDelay", true, time.Second, 0, 0},
+		{"ImmediateWithDelay", true, 500 * time.Millisecond, time.Second, 1},
+		{"NonImmediateWithoutDelay", false, time.Second, 0, 1},
+		{"NonImmediateWithDelay", false, 500 * time.Millisecond, time.Second, 2},
+	} {
+		c := c
```

Comment:
Why this variable copy/shadowing?

Reply:
probably it's a habit left after [scopelint](https://github.com/kyoh86/scopelint) :)

Reply:
I didn't know about that linter :+1:

Reply:
this one is a part of [golangci-lint](https://github.com/golangci/golangci-lint)
we could probably use it if we have a need in linters, however we should consider that they consume too much RAM